### PR TITLE
allow rename table between databases

### LINKED ISSUE #2106: sql: allow renaming tables between databases
`ALTER TABLE test.t RENAME to otherdb.t` will cause syntax error. 
The grammer doesn't allow the new table name contain a database prefix.

### PR DESCRIPTION
**Pull request by wenyong-h**
Fixes #2106.

### CODE REVIEW

**Comment by wenyong-h** [comment]
Fixes #2106.

File: sql/rename.go [hunk 1]
```diff
@@ -83,7 +87,7 @@ func (p *planner) RenameTable(n *parser.RenameTable) (planNode, error) {
 		return nil, err
 	}
 
-	if n.Name.Table() == string(n.NewName) {
+	if n.Name.Table() == n.NewName.Table() {
```
Comment (tamird) [inline sql/rename.go]: the `noop` case requires that the src/dst databases are also identical, so this is not correct

Comment (tamird) [inline sql/rename.go]: please add a test that exercises this case as well

**Comment by tamird** [comment]
LGTM with comments fixed. Welcome to CockroachDB!

File: sql/parser/sql.y [hunk 2]
```diff
@@ -1392,13 +1392,13 @@ rename_stmt:
   {
     $$ = &RenameDatabase{Name: Name($3), NewName: Name($6)}
   }
-| ALTER TABLE relation_expr RENAME TO name
+| ALTER TABLE relation_expr RENAME TO relation_expr
```
Comment (petermattis) [inline sql/parser/sql.y]: I think the `TO` name should be `qualified_name`, not `relation_expr`. The latter rule allows for `ONLY table_name` and `table_name *` which I do not think makes sense in the context of renaming a table.

**Comment by petermattis** [comment]
LGTM

**Comment by wenyong-h** [comment]
Hey, I've adrressed your comments.

**Comment by tamird** [comment]
Nice work!

**Pull request merged**